src/core/backbone_router/local.cpp

```diff
@@ -0,0 +1,246 @@
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements local Backbone Router service.
+ */
+
+#include "local.hpp"
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+
+#include "common/code_utils.hpp"
+#include "common/instance.hpp"
+#include "common/locator-getters.hpp"
+#include "common/logging.hpp"
+#include "common/random.hpp"
+#include "thread/mle.hpp"
+#include "thread/mle_constants.hpp"
+#include "thread/thread_netif.hpp"
+
+namespace ot {
+
+namespace BackboneRouter {
+
+Local::Local(Instance &aInstance)
+    : InstanceLocator(aInstance)
+    , mIsServiceAdded(false)
+    , mState(OT_BACKBONE_ROUTER_STATE_DISABLED)
+    , mSequenceNumber(Random::NonCrypto::GetUint8())
+    , mReregistrationDelay(Mle::kRegistrationDelayDefault)
+    , mMlrTimeout(Mle::kMlrTimeoutDefault)
+{
+    // Primary Backbone Router Aloc
+    memset(&mBackboneRouterPrimaryAloc, 0, sizeof(mBackboneRouterPrimaryAloc));
+
+    mBackboneRouterPrimaryAloc.mPrefixLength       = 128;
+    mBackboneRouterPrimaryAloc.mPreferred          = true;
+    mBackboneRouterPrimaryAloc.mValid              = true;
+    mBackboneRouterPrimaryAloc.mScopeOverride      = Ip6::Address::kRealmLocalScope;
+    mBackboneRouterPrimaryAloc.mScopeOverrideValid = true;
+}
+
+void Local::SetEnabled(bool aEnable)
+{
+    VerifyOrExit(aEnable == (mState == OT_BACKBONE_ROUTER_STATE_DISABLED));
+
+    if (aEnable)
+    {
+        AddService();
+        SetState(OT_BACKBONE_ROUTER_STATE_SECONDARY);
+    }
+    else if (mState == OT_BACKBONE_ROUTER_STATE_PRIMARY)
```

Comment:
Since we have `bbr register` CLI command and `otBackboneRouterRegisterLocal`, it should not be a rare case that `SBBR` has `BBR dataset` in local server data.

Reply:
good point. will enhance.